// @metaplex-foundation/mpl-hydra/dist/src/generated/programs/mplHydra.js
"use strict";
/**
 * This code was AUTOGENERATED using the kinobi library.
 * Please DO NOT EDIT THIS FILE, instead use visitors
 * to add features, then rerun kinobi to update it.
 *
 * @see https://github.com/metaplex-foundation/kinobi
 */
Object.defineProperty(exports, "__esModule", { value: true });
exports.getMplHydraProgramId = exports.getMplHydraProgram = exports.createMplHydraProgram = exports.MPL_HYDRA_PROGRAM_ID = void 0;
const errors_1 = require("../errors");
exports.MPL_HYDRA_PROGRAM_ID = 'hyDQ4Nz1eYyegS6JfenyKwKzYxRsCWCriYSAjtzP4Vg';
function createMplHydraProgram() {
    return {
        name: 'mplHydra',
        publicKey: exports.MPL_HYDRA_PROGRAM_ID,
        getErrorFromCode(code, cause) {
            return (0, errors_1.getMplHydraErrorFromCode)(code, this, cause);
        },
        getErrorFromName(name, cause) {
            return (0, errors_1.getMplHydraErrorFromName)(name, this, cause);
        },
        isOnCluster() {
            return true;
        },
    };
}
exports.createMplHydraProgram = createMplHydraProgram;
function getMplHydraProgram(context, clusterFilter) {
    return context.programs.get('mplHydra', clusterFilter);
}
exports.getMplHydraProgram = getMplHydraProgram;
function getMplHydraProgramId(context, clusterFilter) {
    return context.programs.getPublicKey('mplHydra', exports.MPL_HYDRA_PROGRAM_ID, clusterFilter);
}
exports.getMplHydraProgramId = getMplHydraProgramId;
