// @metaplex-foundation/mpl-hydra/dist/src/generated/instructions/unstake.js
"use strict";
/**
 * This code was AUTOGENERATED using the kinobi library.
 * Please DO NOT EDIT THIS FILE, instead use visitors
 * to add features, then rerun kinobi to update it.
 *
 * @see https://github.com/metaplex-foundation/kinobi
 */
Object.defineProperty(exports, "__esModule", { value: true });
exports.unstake = exports.getUnstakeInstructionDataSerializer = void 0;
const umi_1 = require("@metaplex-foundation/umi");
const serializers_1 = require("@metaplex-foundation/umi/serializers");
const shared_1 = require("../shared");
function getUnstakeInstructionDataSerializer(_context = {}) {
    return (0, serializers_1.mapSerializer)((0, serializers_1.struct)([['discriminator', (0, serializers_1.array)((0, serializers_1.u8)(), { size: 8 })]], { description: 'UnstakeInstructionData' }), (value) => ({
        ...value,
        discriminator: [217, 160, 136, 174, 149, 62, 79, 133],
    }));
}
exports.getUnstakeInstructionDataSerializer = getUnstakeInstructionDataSerializer;
// Instruction.
function unstake(context, input) {
    const signers = [];
    const keys = [];
    // Program ID.
    const programId = context.programs.getPublicKey('mplHydra', 'hyDQ4Nz1eYyegS6JfenyKwKzYxRsCWCriYSAjtzP4Vg');
    // Resolved inputs.
    const resolvedAccounts = {
        member: [input.member, true],
        fanout: [input.fanout, true],
        membershipVoucher: [input.membershipVoucher, true],
        membershipMint: [input.membershipMint, true],
        membershipMintTokenAccount: [
            input.membershipMintTokenAccount,
            true,
        ],
        memberStakeAccount: [input.memberStakeAccount, true],
        instructions: [input.instructions, false],
    };
    (0, shared_1.addObjectProperty)(resolvedAccounts, 'systemProgram', input.systemProgram
        ? [input.systemProgram, false]
        : [
            context.programs.getPublicKey('splSystem', '11111111111111111111111111111111'),
            false,
        ]);
    (0, shared_1.addObjectProperty)(resolvedAccounts, 'tokenProgram', input.tokenProgram
        ? [input.tokenProgram, false]
        : [
            context.programs.getPublicKey('splToken', 'TokenkegQfeZyiNwAJbNbGKPFXCWuBvf9Ss623VQ5DA'),
            false,
        ]);
    (0, shared_1.addAccountMeta)(keys, signers, resolvedAccounts.member, false);
    (0, shared_1.addAccountMeta)(keys, signers, resolvedAccounts.fanout, false);
    (0, shared_1.addAccountMeta)(keys, signers, resolvedAccounts.membershipVoucher, false);
    (0, shared_1.addAccountMeta)(keys, signers, resolvedAccounts.membershipMint, false);
    (0, shared_1.addAccountMeta)(keys, signers, resolvedAccounts.membershipMintTokenAccount, false);
    (0, shared_1.addAccountMeta)(keys, signers, resolvedAccounts.memberStakeAccount, false);
    (0, shared_1.addAccountMeta)(keys, signers, resolvedAccounts.systemProgram, false);
    (0, shared_1.addAccountMeta)(keys, signers, resolvedAccounts.tokenProgram, false);
    (0, shared_1.addAccountMeta)(keys, signers, resolvedAccounts.instructions, false);
    // Data.
    const data = getUnstakeInstructionDataSerializer().serialize({});
    // Bytes Created On Chain.
    const bytesCreatedOnChain = 0;
    return (0, umi_1.transactionBuilder)([
        { instruction: { keys, programId, data }, signers, bytesCreatedOnChain },
    ]);
}
exports.unstake = unstake;
